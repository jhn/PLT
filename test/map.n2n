fn map() -> Void {

	actor: Node = { name: String, age: Int }
	Keanu: Node = actor["Keanu", 35]
	Leo: Node = actor["Leo", 20]

	actedIn: Rel = { role: String }
	neo: Rel = actedIn["Neo"]
	jordan: Rel = actedIn["Jordan"]

	movie: Node = { title: String, year: Int }
	matrix: Node = movie["Matrix", 1999]
	wolf: Node = movie["Wolf", 1994]

	Cast: Graph = (
		Keanu neo matrix,
		Leo jordan wolf
	)

	visited_node: List<Node> = Cast.map(node in {node [+] visited:Bool = true})

	print(visited_node)

}