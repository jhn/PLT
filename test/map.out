[actor{ name = Keanu, age = 35, visited = true }, {Matrix, 1999, true}, {Leo, 20, true}, {Wolf, 1994, true}]
